src/Experimental/RAGTools/generation.jl

```diff
@@ -63,6 +64,45 @@ function build_context(contexter::ContextEnumerator,
     return context
 end
 
+"""
+    build_context(contexter::ContextEnumerator,
+        index::AbstractManagedIndex, candidates::AbstractCandidateWithChunks;
+        verbose::Bool = true,
+        chunks_window_margin::Tuple{Int, Int} = (1, 1), kwargs...)
+
+        build_context!(contexter::ContextEnumerator,
+        index::AbstractManagedIndex, result::AbstractRAGResult; kwargs...)
+
+Dispatch for `AbstractManagedIndex` with `AbstractCandidateWithChunks`.
+"""
+function build_context(contexter::ContextEnumerator,
+        index::AbstractManagedIndex,
+        candidates::AbstractCandidateWithChunks;
+        verbose::Bool = true,
+        chunks_window_margin::Tuple{Int, Int} = (1, 1), kwargs...)
+    ## Checks
+    @assert chunks_window_margin[1] >= 0&&chunks_window_margin[2] >= 0 "Both `chunks_window_margin` values must be non-negative"
+
+    context = String[]
+    for (i, position) in enumerate(positions(candidates))
+        ## select the right index
+        id = candidates isa MultiCandidateChunks ? candidates.index_ids[i] :
+             candidates.index_id
+        index_ = index isa AbstractChunkIndex ? index : index[id]
+        isnothing(index_) && continue
+
+        chunks_ = chunks(candidates)[
+            max(1, position - chunks_window_margin[1]):min(end,
```

Comment:
If I understand well, you cannot use the chunk window extension with Pinecone? You’ll have only the chunks directly returned and nothing more. 

The trick here that we would grab also the surrounding chunks but that wont work here!